src/NexusMods.Paths/Utilities/KnownExtensions.cs

```diff
@@ -1,52 +1,650 @@
+
+    /// <summary>
+    /// Gets the category of a given extension.
+    /// </summary>
+    /// <param name="extension">The extension to get the category for.</param>
+    /// <returns>The <see cref="ExtensionCategory"/> that the extension belongs to.</returns>
+    public static ExtensionCategory GetCategory(this Extension extension)
+    {
+        return ExtensionToCategoryMap.GetValueOrDefault(extension, ExtensionCategory.Unknown);
+    }
+
+    #endregion
+
+    private static readonly Dictionary<Extension, ExtensionCategory> ExtensionToCategoryMap = new()
+    {
+        // Note (Sewer): It'd be nice to source generate this.
+
+        // Archives
+        { _7z, ExtensionCategory.Archive },
+        { _7zip, ExtensionCategory.Archive },
+        { Archive, ExtensionCategory.Archive },
+        { Afs, ExtensionCategory.Archive },
+        { Ba2, ExtensionCategory.Archive },
+        { Bdt, ExtensionCategory.Archive },
+        { Bht, ExtensionCategory.Archive },
+        { Bfs, ExtensionCategory.Archive },
+        { Bsa, ExtensionCategory.Archive },
+        { Bundle, ExtensionCategory.Archive },
+        { Cpk, ExtensionCategory.Archive },
+        { Hkx, ExtensionCategory.Archive },
+        { Nx, ExtensionCategory.Archive },
+        { Oiv, ExtensionCategory.Archive },
+        { Pak, ExtensionCategory.Archive },
+        { Rar, ExtensionCategory.Archive },
+        { Uasset, ExtensionCategory.Archive },
+        { Unitypackage, ExtensionCategory.Archive },
+        { Vpk, ExtensionCategory.Archive },
+        { Wad, ExtensionCategory.Archive },
+        { Xnb, ExtensionCategory.Archive },
+        { Zip, ExtensionCategory.Archive },
+        // ArchiveOfAudio
+        { Awb, ExtensionCategory.ArchiveOfAudio },
+        { Bank, ExtensionCategory.ArchiveOfAudio },
+        { Bnk, ExtensionCategory.ArchiveOfAudio },
+        { Csb, ExtensionCategory.ArchiveOfAudio },
+        { Fsb, ExtensionCategory.ArchiveOfAudio },
+        // ArchiveOfTexture
+        { Txd, ExtensionCategory.ArchiveOfImage },
+        { Tpf, ExtensionCategory.ArchiveOfImage },
+        // Audio
+        { Aac, ExtensionCategory.Audio },
+        { Adx, ExtensionCategory.Audio },
+        { Aix, ExtensionCategory.Audio },
+        { Ahx, ExtensionCategory.Audio },
+        { Fmod, ExtensionCategory.Audio },
+        { Flac, ExtensionCategory.Audio },
+        { Hca, ExtensionCategory.Audio },
+        { Mp3, ExtensionCategory.Audio },
+        { M4a, ExtensionCategory.Audio },
+        { Ogg, ExtensionCategory.Audio },
+        { Wav, ExtensionCategory.Audio },
+        { Xma, ExtensionCategory.Audio },
+        { Xwb, ExtensionCategory.Audio },
+        { Xwm, ExtensionCategory.Audio },
+        // Binary
+        { Tmp, ExtensionCategory.Binary },
+        { Xsb, ExtensionCategory.Binary },
+        // Database
+        { Sqlite, ExtensionCategory.Database },
+        { Esp, ExtensionCategory.Database },
+        { Esm, ExtensionCategory.Database },
```

Comment:
ESP and ESM files aren't databases.

Reply:
What would you classify them as then?

I thought 'Database' would be the best fit as these files are essentially binary (non-text) collections of records, which is very close to what a database is. If you have an alternative suggestion, let me know.

Reply:
I wouldn't classify them at all.

Reply:
In 13bd1b5407a58ca740a3a9e76c7d57f7c5b5a993 I clarified the category to also be 'anything that contains a collection of records'. Hopefully that helps alleviate the situation. If we ever use that category, and don't like how we use it, we can remove it.